architecture-docs/adr-003.md

```diff
@@ -118,19 +119,21 @@ its nodes will be written into the `StoreKV` buffer, which is the buffered key-v
 - The complete block commit processing will be done as follows at the end:
   
   ```rust
-  flush_staking(&mut kv_store!(self), block_height, staking_buffer);
+  flush_staking(&mut kv_store!(self), version, staking_buffer);
   flush_kv(&mut self.kvdb, kv_buffer);  // Write to the disk atomically.
   ```
 
 ### Chain Node State Changes
 
 - `AccountStorage` will be removed from `ChainNodeApp` (as the node storage will be in the dedicated `COL_TRIE_NODE` column).
 
-- In ABCIQuery, the block height from the request will be used to query historical states directly.
+- The block height to root hash mapping in storage will be changed to block height to version number.
+
+  ABCIQuery will convert block height to version number first to query history state.
```

Comment:
Or rather, in that tx-query setting, it'll store the historical `ChainState`s, so it'll lookup chain state at that height and get its version?

Reply:
Yes, but currently the state is only stored optionally.

Reply:
yep -- this abciquery path will be for full nodes serving light clients